main.cpp

```diff
@@ -1,22 +1,217 @@
 #include <iostream>
 #include <SDL2/SDL.h>
+#include <string>
+#include "res_path.h"
+//#include "cleanup.h"
+
+void logSDLError(std::ostream &os, const std::string &msg){
+	os << msg << " error: " << SDL_GetError() << std::endl;
+}
+
+
+SDL_Texture* loadTexture(const std::string &file, SDL_Renderer *ren){
+
+	SDL_Texture *texture = nullptr;
+	
+	SDL_Surface *loadedImage = SDL_LoadBMP(file.c_str());
+
+	if (loadedImage != nullptr){
```

Comment:
Użyj wzorca "early return" (https://www.fluentcpp.com/2018/08/24/how-to-design-early-returns-in-c-based-on-procedural-programming/)

Reply:
Być może to jest lepszy link https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement